src/world/message_server.cpp

```diff
@@ -308,6 +296,46 @@ void message_server_listen(const bool& requestExit)
     }
 }
 
+void cache_zone_settings()
+{
+    const char* query = "SELECT zoneid, zoneip, zoneport, misc FROM zone_settings;";
+    int         ret   = sql->Query(query);
+
+    if (ret == SQL_ERROR)
+    {
+        ShowError("Error loading zone settings from DB. World server will not send messages.");
+        return;
+    }
+
+    // Keep track of the zones, as well as a list of unique ip / port combinations.
+    std::set<uint64> zoneIppSet;
+    std::set<uint64> yellZoneIppSet;
+    zoneSettingsMap = std::unordered_map<uint16, zone_settings_t>(sql->NumRows());
+    while (sql->NextRow() == SQL_SUCCESS)
+    {
+        uint64 ip = 0;
+        inet_pton(AF_INET, (const char*)sql->GetData(1), &ip);
+        uint64 port = sql->GetUIntData(2);
+
+        zone_settings_t zone_settings;
+        zone_settings.zoneid = sql->GetUInt64Data(0);
+        zone_settings.ipp    = ip | (port << 32);
+        zone_settings.misc   = sql->GetUIntData(3);
+
+        zoneIppSet.insert(zone_settings.ipp);
+        if (zone_settings.misc & 1024)
```

Comment:
Presumably there's an enum somewhere to make it clearer what this 1024 is used for